behind-the-veil-siteroot/imports/ui/components/searchBar/searchBar.jsx

```diff
@@ -4,57 +4,244 @@
  * Contributors: Lucas, Nikki
  */
 
-import React from "react";
-import {XMarkIcon} from "@heroicons/react/24/outline";
+import React, {useEffect, useState} from "react";
+import {MagnifyingGlassIcon, XMarkIcon} from "@heroicons/react/24/outline";
 import classNames from "classnames";
 import "./searchBar.css";
 import Input from "../input/Input";
+import Button from "../button/Button";
+import UrlBasePath from "../../enums/UrlBasePath";
+import {useNavigate} from "react-router-dom";
+import {useServices, useUsers} from "../DatabaseHelper";
+import {useSearchSuggestions} from "../util";
 
 /**
  * General Search Bar component for all required instances of the searching in the app.
  * Works with enter button on keyboard (Requires a separate button for users who use mouse click)
  * @param className custom classes for the search input field
- * @param placeholderName the placeholder name for the search input field (i.e. what the user will see if the input field is empty)
- * @param value the value of the input field (whatever the customer inputs)
- * @param onChange the function that occurs when the search input field changes
- * @param handleSubmit the function which dictates what happens when a user submits the value in the search input field
+ * @param defaultType
+ * @param placeholder
+ * @param startingValue
+ * @param suggestionsDown
  * @param searchBarProps encompasses all other props supplied and applies them to the search input field
  */
 const SearchBar = ({
-                       className,
-                       placeholderName,
-                       value,
-                       onChange,
-                       handleSubmit,
+                       classnames,
+                       defaultType = 'services',
+                       placeholder,
+                       startingValue,
+                       suggestionsDown,
                        ...searchBarProps
                    }) => {
+    let navigate = useNavigate();
+
+    // if type is not a valid value, default to services
+    if (defaultType !== 'services' && defaultType !== 'artists') {
+        defaultType = 'services'
+    }
+
+    // creates a state of the input value for the search bar
+    const [inputValue, setInputValue] = useState(startingValue);
+    const [searchType, setSearchType] = useState(defaultType);
+
+    // load data
+    const userFilter = {"profile.type": "artist"}
+    const {isLoading: isLoadingArtists, usersData} = useUsers("all_artists", [], userFilter)
+
+    const serviceFilter = {serviceActive: true}
+    const {isLoading: isLoadingServices, servicesData} = useServices("active_services", [], serviceFilter, true)
+
+    // variables set up for suggestions
+    const [fullSuggestions, setFullSuggestions] = useState([]);
```

Comment:
I think fullSuggestions doesn't even need to be a state. Looks like you're just calculating the suggestions based on searchType (which is a state) and usersData and servicesData. 

Since you're deriving the list of suggestions from an already existing state(s?), you can just do something like
`const fullSuggestions = useSearchSuggestions(searchType, usersData, servicesData)
which is a derived state, and the list of suggestions will stay in sync with those inputs because fullSuggestions will be recalculated whenever searchType's state is changed (this explains derived states well i think https://medium.com/comsystoreply/totally-in-sync-using-derived-state-in-react-applications-98cbd5acaa91)

but basically, if a variable can be solely derived from another state(s), it's better to just derive it instead of creating another state, b/c otherwise you have to always make sure they're in sync => more likely for errors, and setting states unnecessarily causes unnecessary renders as well.

Reply:
changed to useMemo :D thank you